common_test.go

```diff
@@ -81,12 +81,81 @@ func assertCipherSuiteParamsEquals(t *testing.T, a, b CipherSuiteParams) {
 
 func assertDeepEquals(t *testing.T, a, b interface{}) {
 	t.Helper()
-	assert(t, reflect.DeepEqual(a, b), fmt.Sprintf("%+v != %+v", a, b))
+	assertTrue(t, reflect.DeepEqual(a, b), fmt.Sprintf("%+v != %+v", a, b))
 }
 
 func assertSameType(t *testing.T, a, b interface{}) {
 	t.Helper()
 	A := reflect.TypeOf(a)
 	B := reflect.TypeOf(b)
-	assert(t, A == B, fmt.Sprintf("%s != %s", A.Name(), B.Name()))
+	assertTrue(t, A == B, fmt.Sprintf("%s != %s", A.Name(), B.Name()))
+}
+
+// Utilities for parametrized tests
+// Represents the configuration for a given test instance.
+type testInstanceState interface {
+	set(string, string)
+	get(string) string
+}
+
+// This complicated mixin stuff is in case we want to extend this to
+// pass other parameters that aren't srings. Arguably YAGNI.
+type testInstanceStateMixin struct {
```

Comment:
I would just leave this as a `map[string]string` for now. (`type testInstanceState map[string]string`)  If you want to let the test pass other stuff in, you could add an `interface{}` argument to the "public" API that the caller can use to pass context to each test.